adapters/unruly/unruly.go

```diff
@@ -13,136 +13,136 @@ import (
+
+	if response.StatusCode == http.StatusBadRequest {
+		return nil, []error{&errortypes.BadInput{
+			Message: fmt.Sprintf("unexpected status code: %d. Run with request.debug = 1 for more info", response.StatusCode),
+		}}
 	}
-}
 
-func CheckResponse(response *adapters.ResponseData) error {
 	if response.StatusCode != http.StatusOK {
-		return &errortypes.BadServerResponse{
-			Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info", response.StatusCode),
-		}
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("unexpected status code: %d. Run with request.debug = 1 for more info", response.StatusCode),
+		}}
 	}
-	return nil
-}
-
-func convertToAdapterBidResponse(response *adapters.ResponseData, internalRequest *openrtb2.BidRequest) (*adapters.BidderResponse, []error) {
-	var errs []error
 	var bidResp openrtb2.BidResponse
 	if err := json.Unmarshal(response.Body, &bidResp); err != nil {
-		return nil, []error{err}
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("bad server response: %d. ", err),
+		}}
 	}
+
+	var errs []error
 	bidResponse := adapters.NewBidderResponseWithBidsCapacity(5)
+
 	for _, sb := range bidResp.SeatBid {
 		for i := range sb.Bid {
-			bidType, err := getMediaTypeForImpWithId(sb.Bid[i].ImpID, internalRequest.Imp)
-			if err != nil {
-				errs = append(errs, err)
-			} else {
-				b := &adapters.TypedBid{
-					Bid:     &sb.Bid[i],
-					BidType: bidType,
-				}
-				bidResponse.Bids = append(bidResponse.Bids, b)
-			}
+			bidResponse.Bids = append(bidResponse.Bids, &adapters.TypedBid{
+				Bid:     &sb.Bid[i],
+				BidType: getMediaTypeForImp(sb.Bid[i].ImpID, internalRequest.Imp),
+			})
 		}
 	}
 	return bidResponse, errs
 }
 
-func convertBidderNameInExt(imp *openrtb2.Imp) (*openrtb2.Imp, error) {
-	var bidderExt adapters.ExtImpBidder
-	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
-		return nil, err
-	}
-	var unrulyExt openrtb_ext.ExtImpUnruly
-	if err := json.Unmarshal(bidderExt.Bidder, &unrulyExt); err != nil {
-		return nil, err
-	}
-	var impExtUnruly = ImpExtUnruly{Unruly: openrtb_ext.ExtImpUnruly{
-		SiteID: unrulyExt.SiteID,
-		UUID:   unrulyExt.UUID,
-	}}
-	bytes, err := json.Marshal(impExtUnruly)
-	if err != nil {
-		return nil, err
-	}
-	imp.Ext = bytes
-	return imp, nil
-}
-
-func (a *UnrulyAdapter) MakeBids(internalRequest *openrtb2.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
-	if err := CheckResponse(response); err != nil {
-		return nil, []error{err}
+func getMediaTypeForImp(impId string, imps []openrtb2.Imp) openrtb_ext.BidType {
+	mediaType := openrtb_ext.BidTypeBanner
+	for _, imp := range imps {
+		if imp.ID == impId {
+			if imp.Banner != nil {
+				mediaType = openrtb_ext.BidTypeBanner
+			} else if imp.Video != nil {
+				mediaType = openrtb_ext.BidTypeVideo
+			}
```

Comment:
Should we code a bit more defensively and return an error in case a given `impId` doesn't match with any of the `imp.ID`s in the array or if a given imp comes with both a `imp.Banner == nil` and `imp.Video == nil`? Maybe the error that `getMediaTypeForImp` returns could end up in `MakeBids`' unused `errs` array.

Reply:
done, I hope this is what you meant.

Reply:
I did. Thank you